sycl/include/sycl/accessor.hpp

```diff
@@ -3300,4 +3300,34 @@ struct hash<sycl::accessor<DataT, Dimensions, AccessMode, AccessTarget,
   }
 };
 
+template <typename DataT, int Dimensions, sycl::access_mode AccessMode>
+struct hash<sycl::host_accessor<DataT, Dimensions, AccessMode>> {
+  using AccType = sycl::host_accessor<DataT, Dimensions, AccessMode>;
+
+  size_t operator()(const AccType &A) const {
+#ifndef __SYCL_DEVICE_ONLY__
+    // getSyclObjImpl() here returns a pointer to either AccessorImplHost
+    auto AccImplPtr = sycl::detail::getSyclObjImpl(A);
+    return hash<decltype(AccImplPtr)>()(AccImplPtr);
+#endif
```

Comment:
Is the macro guard needed? If so, would we not have a problem on device as the function wouldn't return a value?

Reply:
Thanks @steffenlarsen for the comment.
My attempt to use host_accessor in any command type (parallel_for, single_task) resulted in run-time error. That's why I ruled out possibility of  function ever be called on device. Am I missing something.

Reply:
Even for the accessors that can be used in kernels, the hash functions are returning bogus values. The problem we could have is if somehow the device compilation path pick up the hash function it would potentially cause warnings and/or errors, even if the hash function was never intended to be used on the device.

Eventually, `host_accessor` wouldn't have special handling for device-side compilation, but since it currently has we likely need the `#else` branch here too.